### PR TITLE
Fixed error when combobox has no options

### PR DESCRIPTION
**Pull request by agliga**
## Description
Added a check to see if `this.activeDescendant`. Also added a test to verify that empty combobox events work. (Test failed without this change)

## References
#1059

### CODE REVIEW

File: src/components/ebay-combobox/index.js [hunk 1]
```diff
@@ -130,7 +130,9 @@ module.exports = require('marko-widgets').defineComponent({
         const newValue = this.inputEl.value;
 
         eventUtils.handleTextInput(originalEvent, () => {
-            this.activeDescendant.reset();
+            if (this.activeDescendant) {
+                this.activeDescendant.reset();
```
Comment (ianmcburnie) [inline src/components/ebay-combobox/index.js]: Hmm, by resetting every time a key is pressed, might this cause the active-descendant to be removed in some cases that we dont want? For example, in the "basic filtering" story:

1. focus on the combobox
1. type "b"
1. press down two times to highlight "Basic Offer 1"
1. type "a"

Does active-descendant still remain highlighted on "Basic Offer 1" or does it dissapear? Sorry I can't test it out myself locally right now.

Comment (agliga) [reply src/components/ebay-combobox/index.js]: Ian, this was added because of this PR https://github.com/eBay/ebayui-core/pull/760
If we remove it, then it will cause the double selection when pressing the down arrow.
At the moment, yes the cursor disappears and then the user has to press down arrow again.

Comment (ianmcburnie) [reply src/components/ebay-combobox/index.js]: I'm not talking about removing the autoReset entirely. Just from doing it every time a key is pressed. Although, having said that, the WCAG example does reset it after every key press, but to me it makes sense to leave the highlighted item in place if a keypress did not trigger any filtering (although how often this scenario is likely to occur is another question - it might be rare especially for long lists). I'll do some testing with screen readers tomorrow to see how this change behaves.

https://www.w3.org/TR/wai-aria-practices-1.1/examples/combobox/aria1.0pattern/combobox-autocomplete-list.html

Comment (agliga) [reply src/components/ebay-combobox/index.js]: Overall that could be a good consideration for another PR or another issue which we can optimize the reset functionality, since this code was introduced in that previous PR to prevent the double highlight. However, this PR was just to fix the no options case, which basically takes the resetting of the active descendant and does it only when it is not null. 

One other thing is the issue seems that the highlight stays if we don't call reset, but when you press down, the first item gets highlighted. So ideally we would have to change makeup to support not resetting the cursor position when the input is changed, and then we can remove the reset. 

Here is an example if I completely remove the reset code. I basically type in the dropdown, select an item with my arrow keys and then type another letter which shouldn't force any  additional filtering (I didn't use the filter one in this example, but it also does the same thing),  and then press the down arrow. 
Here is a gif of that 
combobox-reset

Comment (ianmcburnie) [reply src/components/ebay-combobox/index.js]: Ok, thanks Andew. It sounds like resetting is the best way forward for now then. I also notice a couple of probably related bugs in the MIND pattern example, so it's probably worth me revisiting the entire behaviour and seeing if there's an underlying bug in makeup when I get the chance. For now resetting after every input keeps it simple and predictable.
